src/Control/Carrier/ContainerRegistryApi/Authorization.hs

```diff
@@ -144,11 +144,13 @@ getAuthToken ::
   -- | Request For which to Get Authorization Token
   Manager ->
   -- | Manager to use for requests
+  Maybe [Text] ->
+  -- | Content-Types for Accept Header
   Maybe AuthToken ->
   -- | Existing Token (if any)
   m (Maybe AuthToken)
-getAuthToken cred reqAttempt manager token = do
-  let request' = applyAuthForExistingToken $ reqAttempt{method = "HEAD"}
+getAuthToken cred reqAttempt manager accepts token = do
```

Comment:
Previously container registry api (as implemented by index.docker.io) did not require `HEAD` calls to have same `Accept` header as manifest's type (specifically for multi-platform). Although, this is not explicitly noted in registry v2 SPEC, if not provided, it throws 404 error. Note that, other public registry do not exhibit this behaviour so far, but seems to work regardless.